robot/Cumulus/tests/browser/recurring_donations/zedit_effective_date_erd_record.robot

```diff
@@ -0,0 +1,106 @@
+     Set Suite Variable  ${CURRENTVALUE}
+     Set Suite Variable  ${DATE}
+     Set Suite Variable  ${EFFECTIVE_MODIFIED_DATE}
+     Set Suite Variable  ${CURRENTVALUE_EDITED}
+     Set Suite Variable  ${EFFECTIVE_DATE_INITIAL}
+     Set Suite Variable  ${DATE_TO_UPDATE}
+
+     &{contact1_fields}=              Create Dictionary          Email=rd2tester@example.com
+     &{recurringdonation_fields} =	 Create Dictionary          Name=ERDTest1
+     ...                                                         npe03__Installment_Period__c=Monthly
+     ...                                                         npe03__Amount__c=100
+     ...                                                         npe03__Open_Ended_Status__c=${TYPE}
+     ...                                                         Status__c=Active
+     ...                                                         Day_of_Month__c=${DAY_OF_MONTH}
+     ...                                                         InstallmentFrequency__c=${FREQUENCY}
+     ...                                                         PaymentMethod__c=${METHOD}
+
+     Setupdata   contact             ${contact1_fields}          recurringdonation_data=${recurringdonation_fields}
+
+
+*** Variables ***
+${FREQUENCY}  1
+${DAY_OF_MONTH}  15
+${AMOUNT_TO_UPDATE}  150
+${METHOD}  Credit Card
+${TYPE}    Open
+
+*** Test Cases ***
+
+Edit An Enhanced Recurring donation record of type open
+    [Documentation]               After creating an open recurring donation using API, The test ensures that when the record
+     ...                          is edited and effective date is updated to next month , Verifies that future schedule is shown
+     ...                          with the right values reflected. Verifies the current year and future year schedules are updated
+     ...                          Verifies the opportunity status
+    [tags]                                   unstable               W-041167            feature:RD2
+
+    Go To Page                               Details
+    ...                                      npe03__Recurring_Donation__c
+    ...                                      object_id=${data}[contact_rd][Id]
+    ${rd_id}                                 Save Current Record ID For Deletion       npe03__Recurring_Donation__c
+    Validate Field Values Under Section
+    ...                                      Amount=$100.00
+    ...                                      Status=Active
+    # Validate the fields under Current Schedule card
+    Validate Field Values Under Section      Current Schedule
+    ...                                      Amount=$100.00
+    ...                                      Payment Method=Credit Card
+    ...                                      Effective Date=${effective_date_initial}
+    ...                                      Installment Period=Monthly
+    ...                                      Day of Month=15
```

Comment:
just curious, since u r already creating this data with api, why do u again need to validate same data via UI?

Reply:
please correct the global variables to be capital in the test case as well

Reply:
Corrected the global variables for all the references to be uniform.

This test case aims at the validation of some fields and UI component rendering before and after the update of a certain field. So the steps were
1) Create Rd using backend - verify that all the UI elements and the components are showing along with the calculations for the current year and next year's donations.
2) Update the field "Effective Date", navigate to the same record and validate that a new UI component gets displayed along with the updated calculations and edited date.